new-log-viewer/src/contexts/StateContextProvider.tsx

```diff
@@ -79,14 +81,25 @@ const StateContextProvider = ({children}: StateContextProviderProps) => {
         const {code, args} = ev.data;
         console.log(`[MainWorker -> Renderer] code=${code}`);
         switch (code) {
-            case WORKER_RESP_CODE.PAGE_DATA:
+            case WORKER_RESP_CODE.PAGE_DATA: {
                 setLogData(args.logs);
-                setLogLines(args.lines);
-                setLogEventNum(args.startLogEventNum);
+                setBeginLineNumToLogEventNum(args.beginLineNumToLogEventNum);
+                const logEventNums = Array.from(args.beginLineNumToLogEventNum.values());
+
+                // Explicit cast since typescript thinks the last item of `logEventNums` can be
```

Comment:
Hmm... Can `logEventNums` be an empty array due to programming errors / when there is no log event in the file?

Reply:
True. In that case, should we set the log event num to 1 (rather than logging an error as we did previously)?

Reply:
I agree